### PR TITLE
add condition for core courses

### PR DESCRIPTION
**Pull request by joooooooooooooooooooooooooooooooooooosh**

### CODE REVIEW

File: backend/algorithms/objects/user.py [hunk 1]
```diff
@@ -21,6 +21,7 @@ def __init__(self, data = None):
         self.program: str = None
         self.specialisations: dict[str, int] = {}
         self.year: int = 0
+        self.core_courses: list[str] = []
```
Comment (Peedee2002) [inline backend/algorithms/objects/user.py:24]: Is there currently a way to populate this?

**Review by Peedee2002 — CHANGES_REQUESTED** [review]
Nice, but we are also gonna need some tests for user.has_taken_cores()

This can be pretty simple because the code isn't complex, but yea. Also, make sure that the cores field can be initialised properly when passing a user json. I can't tell if youre currently handling that case.

File: backend/algorithms/objects/user.py (hunk 1 above)
Comment (joooooooooooooooooooooooooooooooooooosh) [reply backend/algorithms/objects/user.py:24]: no that'll be in the next stage where we add this functionality to the frontend

Comment (joooooooooooooooooooooooooooooooooooosh) [reply backend/algorithms/objects/user.py:24]: this pr is just outlining the structure etc

**Comment by joooooooooooooooooooooooooooooooooooosh** [comment]
> Nice, but we are also gonna need some tests for user.has_taken_cores()

do you mean has_taken_course() or completed_core()?
has_taken_course() was already there we just changed it to utilize hussain's marks pr

**Comment by Peedee2002** [comment]
> > Nice, but we are also gonna need some tests for user.has_taken_cores()
> 
> do you mean has_taken_course() or completed_core()? has_taken_course() was already there we just changed it to utilize hussain's marks pr

Ah yes sorry I meant both. Just 1 test for a course with less than 50% and another few tests for core.

File: backend/algorithms/objects/user.py (hunk 1 above)
Comment (Peedee2002) [reply backend/algorithms/objects/user.py:24]: Ah fair

**Pull request merged**